### PR TITLE
EHN: data catalog unit attributes

### PR DESCRIPTION
**Pull request by Tjalling-dejong**
See issue #327

**Done**
- I implemented the functionality that allows for setting the attributes of variables in RasterDatasetAdapter.  The datacatalog allows an attrs argument that should refer to the variables in the dataset and can be given attribute data, for instance unit or long name of the variable.  
- Implement the same functionality for the GeoDatasetAdapter
- Implement the same functionality for the DataframeAdapter
- And GeoDataframeAdapter

### CODE REVIEW

File: hydromt/data_adapter/data_adapter.py
Comment (savente93) [inline hydromt/data_adapter/data_adapter.py:195]: Can we remove this commented code?

File: tests/test_data_adapter.py
Comment (savente93) [inline tests/test_data_adapter.py:240]: If we multiply like this, should the unit name name also be adjusted? (no is a perfectly acceptable answer here, I'm genuinely not sure)

File: hydromt/vector.py
Comment (savente93) [inline hydromt/vector.py:32]: Is it okay to keep the placeholder like this, or should we decide on a value?

File: .pre-commit-config.yaml
Comment (savente93) [inline .pre-commit-config.yaml:19]: Be sure to fix this merge conflict before you merge. pulling main should be ok.

**Review by savente93 — COMMENTED** [review]
Very nice work Tjalling, code looks to be very well tested. Just a few comments about polish, but most of it looks good. Will be happy to approve once comments are resolved.

File: hydromt/data_adapter/data_adapter.py
Comment (DirkEilander) [inline hydromt/data_adapter/data_adapter.py:224]: Why is this needed? basically have decided that "driver_kwargs" replaces the old "kwargs" right?

**Review by DirkEilander — CHANGES_REQUESTED** [review]
Indeed nice work @Tjalling-dejong! The code looks good to me. Regarding the docs I have two remarks
- I noticed some errors we made copy-pasting the docstrings for the get_data methods. Some of the docstrings info applies to other DataAdapters. These should be fixed. 
- Could you still add a description of the new `attrs` keyword
- Could you add a note to `driver_kwargs` and `attrs` that these keywords have been changed / added that from version 0.7.2

both changes re in docs/user_guide/data_prepare_cat.rst I think

File: tests/test_data_adapter.py
Comment (Tjalling-dejong) [reply tests/test_data_adapter.py:250]: Yes, I made a new issue (#431) for it. I'll remove the comments.

File: hydromt/data_adapter/data_adapter.py
Comment (Tjalling-dejong) [reply hydromt/data_adapter/data_adapter.py:195]: Yes, I removed it.

File: tests/test_data_adapter.py
Comment (Tjalling-dejong) [reply tests/test_data_adapter.py:240]: In theory yes, but we are only testing the multiplication of the units. Changing the unit name is already covered in other tests.

File: tests/test_data_adapter.py
Comment (Tjalling-dejong) [reply tests/test_data_adapter.py:390]: It can have any name tbh, its only dummy data :)

File: hydromt/vector.py
Comment (Tjalling-dejong) [reply hydromt/vector.py:32]: I removed it

File: .pre-commit-config.yaml
Comment (Tjalling-dejong) [reply .pre-commit-config.yaml]: fixed it

**Review by DirkEilander — APPROVED** [review]
I've made a few final changes, mainly to the docs and tests as well as adding a nodata property to the GeoDataset class. (vector accessor). Can you have a look at my final changes and merge if you're happy @Tjalling-dejong?

File: hydromt/data_adapter/dataframe.py
Comment (DirkEilander) [inline hydromt/data_adapter/dataframe.py]: DataFrames do not have a CRS property as it does not have spatial data

**Pull request merged**
